code/de_9im.R

```diff
@@ -50,27 +51,49 @@ de_9im = function(x,
   if(output == "character") {
     res = unlist(funs)[res]
   }
+  res_text2 = ""
   if(include_relate) {
     relation = sf::st_relate(x, y)
-    relate_text = paste0(" \nDE-9IM string: \n", relation) 
-    res = c(res, relate_text)
+    res_text2 = paste0(" \nDE-9IM string: \n", relation) 
   }
   if(plot) {
-    res_text = paste(res, collapse = collapse)
+    res_text1 = paste(res, collapse = collapse)
     message("Object x has the following spatial relations to y: ", res_text)
-    res = de_9im_plot(xy, label = res_text)
+    if (tmap){
+      res = de_9im_plot2(xy, label1 = res_text1, label2 = res_text2)
+    } else {
+      res = de_9im_plot(xy, label1 = res_text1, label2 = res_text2)
+    }
   }
   res
 }
 
-de_9im_plot = function(xy, label = "test", alpha = 0.5, show.legend = FALSE, x = 0.1, y = 0.95, theme = ggplot2::theme_void()) {
+de_9im_plot = function(xy, label1 = "test", label2 = "",
+                       alpha = 0.5, show.legend = FALSE, x = 0.1, y = 0.95, 
+                       theme = ggplot2::theme_void()) {
   require("ggplot2", quietly = TRUE)
   # browser()
-  ggplot(xy) + geom_sf(aes(fill = Object), alpha = alpha, show.legend = show.legend) +
-    annotate("text", x = 0.1, y = 0.95, label = label, hjust = "left", vjust = "top") +
+  ggplot(xy) + 
+    geom_sf(aes(fill = Object), alpha = alpha, show.legend = show.legend) +
+    annotate("text", x = 0.1, y = 0.95, label = label1, hjust = "left", vjust = "top") +
+    annotate("text", x = 0.1, y = 0.1, label = label2, hjust = "left", vjust = "bottom", 
+             fontface = "italic") +
     theme
 }
 
+de_9im_plot2 = function(xy, label1 = "test", label2 = "",
+                       alpha = 0.5, show.legend = FALSE, x = 0.1, y = 0.95, 
+                       theme = ggplot2::theme_void()) {
+  require("tmap", quietly = TRUE)
+  # browser()
+  tm_shape(xy) +
+    tm_polygons("Object", legend.show = FALSE, alpha = alpha,
+                palette = c("#E36939", "#6673E3")) +
+    tm_credits(label1, position = c("left", "top")) +
```

Comment:
Idea: we could add #geocompr to the credits.

Reply:
@Robinlovelace -- we can consider adding #geocompr credits to some of the figures in the book explaining common ideas. 

Pros: better attribution, etc.
Cons: maybe, it won't look good in the book (have you ever seen such credits in a book?)

On the other hand, we could select a few figures and create two versions -- html ones with credits and pdf ones without...